src/ui/DynamicFacet/DynamicFacet.ts

```diff
@@ -410,6 +402,39 @@ export class DynamicFacet extends Component {
     this.dynamicFacetQueryController.enableFreezeFacetOrderFlag();
   }
 
+  /**
+   * Returns the facet's state metadata
+   */
+  public get analyticsFacetState(): IAnalyticsDynamicFacetMeta[] {
+    return this.values.activeFacetValues.map(facetValue => facetValue.analyticsMeta);
+  }
+
+  /**
+   * Logs a `Search` usage analytics event containing the metadata of the facet.
+   * @param actionCause The cause of the event.
+   * @param facetMeta The metadata of the `DynamicFacet` component.
+   */
+  public logAnalyticsEvent(actionCause: IAnalyticsActionCause, facetMeta?: IAnalyticsDynamicFacetMeta) {
```

Comment:
Do all those method related to analytics need to be public/documented ?

I'm not convinced it makes a lot of sense that expose those publicly and document them to be usable by a developers, since it's kinda "internal" logic for the component.

As in, I would expect a developer to be able to select value in the facet, show more, etc... (so, all end users action should be exposed and usable through code only), but "logging analytics" is something that is done automatically by the component as a collateral of other actions.

This is also generally not something that is documented for other components.

Reply:
I agree what Olivier says; I don't think we should expose those methods in the public documentation.